examples/retrieve_audit_events.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+#  coding: utf-8
+#  Copyright (c) 2016, 2018, Oracle and/or its affiliates. All rights reserved.
+
+#  This script retrieves all audit logs across an OCI Tenancy.
+#  for a timespan defined by start_time and end_time.
+#  This sample script retrieves Audit events for last 5 days.
+#  This script will work at a tenancy level only.
+
+
+import oci
+from datetime import date
+import datetime
+from dateutil.relativedelta import relativedelta
+
+
+def get_regions(identity):
+    '''
+    To retrieve the list of all available regions.
+    '''
+    list_of_regions = []
+    list_regions_response = identity.list_regions()
+    for r in list_regions_response.data:
+        list_of_regions.append(r.name)
+    return list_of_regions
+
+
+def get_compartments(identity, tenancy_id):
+    '''
+    Retrieve the list of compartments under the tenancy.
+    '''
+    compartment_ocids = []
+    #  Store tenancy id as the first compartment
+    compartment_ocids.append(tenancy_id)
+    list_compartments_response = oci.pagination.list_call_get_all_results(
+        identity.list_compartments,
+        compartment_id=tenancy_id).data
+    for c in list_compartments_response:
+        compartment_ocids.append(c.id)
+    return compartment_ocids
+
+
+def get_audit_events(audit, compartment_ocids, start_time, end_time):
+    '''
+    Get events iteratively for each compartment defined in 'compartments' array
+    for the region defined in "config['region']".
+    '''
+    list_of_audit_events = []
+    for c in compartment_ocids:
+        #  To separate results by compartments use print here.
+        list_events_response = oci.pagination.list_call_get_all_results(
+            audit.list_events,
+            compartment_id=c,
+            start_time=start_time,
+            end_time=end_time).data
+
+        #  Results for a compartment 'c' for a region defined
+        #  in 'audit' object.
+        #  Events can be filtered here based on some constraints.
+        #  For example, to filter write events make use of
+        #  the following condition
+        #  list_events_response.data.request_action != 'GET'
```

Comment:
This comment is incorrect as `list_events_response.data` is an array so so the statement would fail at runtime. 

Additionally, if you want to tell customers about filtering, it's not going to be the most efficient if they eager load the entire result set and then filter - it would be better if they use one of the methods in `oci.pagination` which returns a generator and results can be filtered as they are available/requested rather than all at once at the end.

If you wish to demonstrate this scenario, I would recommend having a separate method which shows customers how to do this properly rather than just as a comment in the code.